cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/preview/PreviewDatasetFramework.java

```diff
@@ -148,6 +149,13 @@ public void addInstance(String datasetTypeName, DatasetId datasetInstanceId,
   }
 
   @Override
+  public void addInstance(String datasetTypeName, DatasetId datasetInstanceId,
+                          DatasetProperties props,
+                          @Nullable KerberosPrincipalId ownerPrincipal) throws DatasetManagementException, IOException {
+    throw new UnsupportedOperationException("Creating dataset instance with owner is not supported");
```

Comment:
In PreviewMode we don't support UGI/Kerberos that is why we don't want to support creating things with an owner.